src/util.h

```diff
@@ -346,6 +346,21 @@ std::string CopyrightHolders(const std::string& strPrefix);
  */
 int ScheduleBatchPriority(void);
 
+/**
+ * Implementation of Hash named requirement for a byte vector. This may be used the hash function in
+ * std::unordered_set or std::unordered_map over std::vector<unsigned char>. Internally, a random
+ * instance of SipHash-2-4 is used.
+ */
+class ByteVectorHash
+{
+private:
+    uint64_t m_k0, m_k1;
+
+public:
+    ByteVectorHash();
+    size_t operator()(const std::vector<unsigned char>& input) const;
+};
```

Comment:
I don't think `util.h` is the best place for a generic utility data structure such as this, it's more for assorted operating system functions

Reply:
@laanwj What would be the right place? A new file `utiltypes.{h,cpp}`? Or `util/types.{h,cpp}`?

Reply:
or maybe `support/bytevectorhash.h` ?
unless there's a strong need to group this with other things